lib/msf/core/post/file.rb

```diff
@@ -159,6 +161,8 @@ def file?(path)
       stat = session.fs.file.stat(path) rescue nil
       return false unless stat
       return stat.file?
+    elsif session.type == 'powershell'
+      return cmd_exec("(Get-Item \"#{path}\").PSIsContainer")&.include?("False")
```

Comment:
Is there any advantage to using this technique vs `Test-Path \"#{path}\" -PathType leaf`? The docs seem to imply that PathType leaf will check if it's a file where as PathType container will check if it's a directory or registry key. See: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/test-path?view=powershell-7.1

Reply:
Both seem to be working fyn but `Test-Path \"#{path}\" -PathType leaf` looks like a better command for testing files specifically :+1:

Reply:
updated and tested!
```
>> file?('C:\\')
=> false
>> exists?('C:\\')
=> true
>> file?('windows_powershell_reverse.exe')
=> true
```